joomla/joomla-cms
# Possibility to change sender of com_contact

### Summary of Changes

This commit enables site owners to choose whether they want to 
a) receive messages from contact form submitters sent from the Joomla! system email address and name or 
b) from the submitters email address and name.

- One new parameter is added.
- Parameter is set to 0 by default for B/C.
- XML-Parameter useglobal='true' is used where required.
- Parameters are implemented in all BE and FE forms.
- Language strings are implemented in english language files (no native speaker!).
- Email copies to the submitter will still be sent with the current system data.
- PHP doc block is updated

### Testing Instructions

**Case 1:**

Testing Instructions 
Apply the patch and test the contact form without changing any settings in advance.
Send email in frontend with checked "email copy".

Expected result: 
No change in behavior. 
Admin will receive contact form emails from website system email an name.
Submitter will receive contact form emails from website system email an name.

Actual result:
Same like expected.

**Case 2:**

Testing Instructions 
Change the new parameter "Sender Type" in the main com_contact settings to "Submitter Data".
Don't change a parameter in the specific contact.
Send email in frontend with checked "email copy".

Expected result: 
Admin will receive contact form emails from **submitter email an name**.
Submitter will receive contact form emails from **website system email an name**.

Actual result:
Admin **and** Submitter getting messages with Joomla! system email and name,

**Case 3:**

Testing Instructions 
Change the new parameter "Sender Type" in the specific com_contact contact settings to "Website Data".
Don't change the parameter in the main com_contact settings again.
Send email in frontend with checked "email copy".

Expected result: 
Admin will receive contact form emails from **website system email an name**.
Submitter will receive contact form emails from **website system email an name**.

Actual result:
See case 1.


### Documentation Changes Required

One new parameter maybe needs to be described.





### Note: 
Besides that the commit simply changes the ordering of two line code at around line 186 of /com_contact/controllers/contact.php to use the same ordering like a few lines above and as in the setSender() parameters below: name after mail address.
